sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -417,7 +418,8 @@ def _receive(self, max_message_count=None, timeout=None):
             self._open()
 
             amqp_receive_client = self._handler
-            received_messages_queue = amqp_receive_client._received_messages
+            received_messages_queue = amqp_receive_client._received_messages # type: ignore[attr-defined]
+            amqp_receive_client._running_iter = False # type: ignore[attr-defined]
```

Comment:
This is something I am not sure about, I use this boolean to show if we are using the iterator or regular receive. If you are receiving normally within an iterator for loop and you sleep, the iterator will time out, however you are still receiving messages so that should not happen. This boolean is a way to reset the last_activity_stamp so that your iterator doesn't time out when receiving via the normal _receive() method. 

uamqp did this a different way, but my thought was that this would be less intrusive into the current design?

Reply:
In pyamqp, we can prevent different receive methods being used together without needing boolean state flags (e.g. locks etc). Would that be useful?
If would make the pyamqp code safer/more resilient - however I'm not sure if it would break back compat with uamqp SB - I can't recall if that supported simultaneously receiving different ways